### ISSUE TITLE
The "New collection" button should be disabled when viewing a collection for which I don't have curate access

### ISSUE THREAD

**Issue by mazameli** [metabase/metabase#14115]
If I'm a user who doesn't have Curate permissions for the collection that I'm currently viewing, I can see and click the "New collection" button. 

There is the possibility that I'm clicking this and trying to create a collection in a _different_ collection than the one I'm viewing, but it feels weird because the dropdown rightly defaults to suggesting that I create the new collection in the collection I'm currently looking at — but I don't have permission to save it there.

Should we hide or disable this button on collections that you only have View access to? That means you'd have to navigate to a collection that you have Curate access to first before you can create a subcollection inside of it, but that sounds somewhat expected to me.

**tl;dr: my current bias is we should ~hide or~ disable the button if you don't have curate access for a given collection.** Thoughts @kdoh?

<img width="703" alt="image" src="https://user-images.githubusercontent.com/2223916/102530689-2f4ef180-4056-11eb-9a8f-625ad228b49d.png">

**Comment by kdoh** [comment]
@mazameli Yeah I would agree.

**Comment by mazameli** [comment]
Here's how it could look in a disabled state, with `text-light` at 25%:

<img width="1275" alt="image" src="https://user-images.githubusercontent.com/2223916/102641171-7ac5d600-4110-11eb-8dcd-4ed9e35a9574.png">
